pcsx2/DEV9/ATA/ATA_State.cpp

```diff
@@ -66,6 +66,17 @@ int ATA::Open(const std::string& hddPath)
 		return -1;
 	}
 
+	// Open and read the content of the hddid file
+	std::string hddidPath = Path::ReplaceExtension(hddPath, "hddid");
+	std::optional<std::vector<u8>> fileContent = FileSystem::ReadBinaryFile(hddidPath.c_str());
+
+	if (fileContent.has_value() && fileContent.value().size() <= sizeof(sceSec))
+	{
+		// Copy the content to sceSec
+		memset(sceSec, 0, sizeof(sceSec));
```

Comment:
Move this to before the if statement
sceSec isn't always cleared when a new HDD file is selected, which could lead to keeping the hddid file in memory when switching from a HDD image with a hddid, to one which does not.

Reply:
done